modules/lti-service-impl/src/main/java/org/opencastproject/lti/service/impl/LtiServiceImpl.java

```diff
@@ -246,60 +250,84 @@ public void upsertEvent(
       throw new RuntimeException("No workflow configured, cannot upload");
     }
     try {
-      MediaPackage mp = ingestService.createMediaPackage();
+      final EventHttpServletRequest r = new EventHttpServletRequest();
+      final MediaPackage mp = ingestService.createMediaPackage();
       if (mp == null) {
         throw new RuntimeException("Unable to create media package for event");
       }
+
       if (captions != null) {
-        final MediaPackageElementFlavor captionsFlavor = new MediaPackageElementFlavor("captions", "vtt+en");
-        final MediaPackageElementBuilder elementBuilder
-            = MediaPackageElementBuilderFactory.newInstance().newElementBuilder();
+        final MediaPackageElementFlavor captionsFlavor = new MediaPackageElementFlavor(
+            "captions", captionFormat + "+" + captionLanguage
+        );
+        final MediaPackageElementBuilder elementBuilder =
+            MediaPackageElementBuilderFactory.newInstance().newElementBuilder();
         final MediaPackageElement captionsMpe = elementBuilder
                 .newElement(MediaPackageElement.Type.Attachment, captionsFlavor);
-        captionsMpe.setMimeType(mimeType("text", "vtt"));
-        captionsMpe.addTag("lang:en");
+        if ("dfxp".equals(captionFormat)) {
+          captionsMpe.setMimeType(mimeType("application", "xml"));
+        } else {
+          captionsMpe.setMimeType(mimeType("text", captionFormat));
+        }
+        captionsMpe.addTag("lang:" + captionLanguage);
         mp.add(captionsMpe);
         final URI captionsUri = workspace
                 .put(
                         mp.getIdentifier().toString(),
                         captionsMpe.getIdentifier(),
-                        "captions.vtt",
+                        "captions." + captionFormat,
                         new ByteArrayInputStream(captions.getBytes(StandardCharsets.UTF_8)));
         captionsMpe.setURI(captionsUri);
       }
 
-      JSONArray metadataJsonArray = (JSONArray) new JSONParser().parse(metadataJson);
-
-      final EventCatalogUIAdapter adapter = getEventCatalogUIAdapter();
+      r.setMediaPackage(mp);
+      final MetadataList metadataList = new MetadataList();
+      final MediaPackageElementFlavor flavor = new MediaPackageElementFlavor("dublincore", "episode");
+      final EventCatalogUIAdapter adapter = catalogUIAdapters.stream()
+          .filter(e -> e.getFlavor().equals(flavor)).findAny().orElse(null);
+      if (adapter == null) {
+        throw new RuntimeException("no adapter found");
+      }
```

Comment:
Not entirely sure about the syntax but there should be something like:

```suggestion
          .filter(e -> e.getFlavor().equals(flavor))
          .findAny()
          .orElseThrow(() -> new RuntimeException("no adapter found"));
```